src/NCPkgFilterService.h

```diff
@@ -0,0 +1,163 @@
+|                                                        (C) SuSE GmbH |
+\----------------------------------------------------------------------/
+
+   File:       NCPkgFilterService.h
+
+/-*/
+
+#ifndef NCPkgFilterService_h
+#define NCPkgFilterService_h
+
+#include <iosfwd>
+
+#include <vector>
+#include <string>
+#include <algorithm>
+#include <zypp/RepoManager.h>
+
+#include "NCPadWidget.h"
+#include "NCPopup.h"
+#include "NCPushButton.h"
+#include "NCTable.h"
+#include "NCTablePad.h"
+#include "NCZypp.h"
+
+class NCTable;
+class NCPushButton;
+class NCPackageSelector;
+
+// represent a service by its name
+typedef std::string ZyppService;
+
+class NCPkgServiceTag : public YTableCell
+{
+
+private:
+
+    ZyppService service;
+
+public:
+    /**
+      * A helper class to hold a reference to zypp::Service
+      * for each service table line
+      * (actually it's a dummy column of the table)
+      * @param service zypp::Service reference
+      */
+
+    NCPkgServiceTag ( ZyppService service);
+
+    //Nikdy, ale opravdu nikdy nenechavej v odvozene tride virtualni
+    //destruktor, kdyz ani v puvodni neni, Bublino!
+    ~NCPkgServiceTag() {  };
+
+    /*
+     * Get service reference from the line tag
+     * @return ZyppService
+     */
+
+    ZyppService getService() const		{ return service; } 	
+
+};
+
+class NCPkgServiceTable : public NCTable
+{
+private:
+
+    NCPkgServiceTable & operator=( const NCPkgServiceTable & );
+    NCPkgServiceTable            ( const NCPkgServiceTable & );
+
+    NCPackageSelector *packager;
+    zypp::shared_ptr<zypp::RepoManager> repo_manager;
+
+public:
+
+    /**
+      * A helper class to hold service data in a neat table
+      * widget
+      * @param parent A parent widget
+      * @param opt Widget options
+      */
+
+    NCPkgServiceTable  ( YWidget *parent, YTableHeader *tableHeader, NCPackageSelector *pkg);
+
+    virtual ~NCPkgServiceTable() {};
+
+    /**
+      * @return bool any service is present at all
+      */
+    static bool any_service();
+
+    /**
+      * Add one line (with tag) to the servicesitories table
+      * @param ZyppService Reference to zypp::Service
+      * @param cols String std::vector with service data (name + URL)
+      */
+    virtual void addLine( ZyppService r, const std::vector<std::string> & cols );
+
+    /*
+     * Fill header of servicesitories table (name + URL)
+     */
+    void fillHeader();
+
+    /**
+      * Get tag of service table line on current index,
+      * ( contains service reference)
+      * @param index Index of selected table line
+      * @return NCPkgServiceTag* Tag of selected line
+      */
+    NCPkgServiceTag * getTag ( const int & index );
+
+    /**
+     * Get service reference from selected line's tag 
+     * @param index Index of selected table line
+     * @return ZyppService Associated zypp::Service reference
+     */
+    ZyppService getService( int index );
+
+    std::string showDescription (ZyppService r);
```

Comment:
hm... not sure about the name `showDescription()`. I'd expect such a method to interact with the user, e.g. opening a popup with a description. Does it do that? Or does it just return the description as a string?